cpp/README.md

```diff
@@ -261,7 +265,7 @@ The optional `gandiva` libraries and tests can be built by passing
 `-DARROW_GANDIVA=on`.
 
 ```shell
-cmake .. -DARROW_GANDIVA=on
+cmake .. -DARROW_GANDIVA=ON -DARROW_BUILD_TESTS=ON
```

Comment:
Should this be `-DARROW_GANDIVA_BUILD_TESTS=ON`?

Reply:
Indeed. Will fix